### ISSUE TITLE
Breaking change: Unable to load / play videos via media queries

### ISSUE THREAD

**Issue by PaulHale** [ampproject/amphtml#30573]
** Breaking Change

It used to be possible to play specific AMP WebStory videos dependent upon media queries. We were loading smaller videos for mobile devices and larger videos for desktop visitors. This was all working perfectly but now the videos won't play with a message saying "Video failed to play(!)"

## What's the issue?
We deployed and it worked.
You broke something (no code changes our end)
Now the videos won't play.

## How do we reproduce the issue?
Visit our site https://www.domainscanners.com
The videos also no longer play in our dev environment. 

## What browsers are affected?
All browsers

## Original HTML which originally worked
`  
<amp-story-page id="p1-cloud-services" auto-advance-after="cloud-services">
          
	  <amp-story-grid-layer template="fill">

		 <amp-video
		    id="cloud-services" 
		    media="(max-width: 720px)"
		    src="/assets/cloud-services-1-720.mp4"
		    width="720" height="1280"
		    layout="responsive"
		    poster="assets/cloud-services-poster-1-720.jpg"
		    autoplay>
	  </amp-video>

		<amp-video
		    id="cloud-services" 
		    media="(min-width: 721px)"
		    src="/assets/cloud-services-1-1080.mp4"
		    width="1920" height="1080"
		    layout="fill"
                    poster="assets/cloud-services-poster-1-1080.jpg" 
                    autoplay>
		</amp-video>
         
    </amp-story-grid-layer>

  </amp-story-page>
`

Please tell us how we can fix this issue.

**Comment by gmajoulet** [comment]
Hey Paul, thank you for creating this ticket.

We dropped the support for selecting which video to play through media queries three months ago (https://github.com/ampproject/amphtml/pull/29233). It allowed us to streamline our video playback pipeline and reduce the error rate to under 0.1%, and this specific fix improved the page load time (LCP) by 15%.

I'm sorry we missed your website and thought no one was using this feature. I didn't think of this `<amp-video media="..."` implementation and only checked other use cases that better work with Stories.

As a replacement, we introduced an adaptive bitrate solution that picks the best video source for an `<amp-video>`. It is not very well documented, but you can find the API and how to use it in this ticket: https://github.com/ampproject/amphtml/issues/29496. I understand it's a bit different from what you were using as you can't force a resolution for a given device, but it will ensure videos play smoothly for all your users.

I'm closing this ticket but feel free to reach out to us if you need any help setting up the adaptive bitrate solution.

**Comment by PaulHale** [comment]
I spent weeks getting our videos to look good on both mobile and desktop. 
Are you now telling me that was a complete waste of my time?

#29496 is all about bitrate. What about video dimensions? 
Are we supposed to only target mobile videos now and not videos targeted at desktop?
I thought the Accelerated "Mobile" Pages ancroym was dropped?
How do I escalate this?

Please advise.

**Comment by PaulHale** [comment]
The more I think about this the more furious I get. 

There should have been an API bump to introduce this change. You can't go around introducing breaking changes and simply state that you're "sorry you missed our site" and "didn't think anyone was using this feature". 
This is totally unaceptable.

Again, please let me know how we can escalate this.

**Comment by gmajoulet** [comment]
> I spent weeks getting our videos to look good on both mobile and desktop.
> Are you now telling me that was a complete waste of my time?

Absolutely not a waste of your time. You spent time generating different transcode quality for your video and it's actually exactly what we recommend doing in #29496.

> #29496 is all about bitrate. What about video dimensions?
> Are we supposed to only target mobile videos now and not videos targeted at desktop?

We spent a lot of time monitoring and optimizing the video experience that was, let's say it, pretty broken. We got the playback error rate down from up to ~60% to less than 0.1%, reduced the joint latency (time to start playing) by a lot, and made the rebuffer events less likely to occur across a full story. All in all this made your website better.
I understand that you used to be able to select a specific resolution and now you can't. As a framework we have been monitoring video performance for millions of sessions and grew a lot more opinionated on what we think is a good experience. We decided to favor the video playback over its quality.
With our new solution, you can't force one or the other resolution to play on a specific device, but you can ensure that any device will play the best quality possible while maintaining a great video playback experience. We think having your video play reliably is what matters the most.
We made these decisions based on metrics and think your website will benefit from it too.

cc @kristoferbaxter @mrjoro on how to escalate.

**Comment by PaulHale** [comment]
I created two versions of every video for our AMP stories, over fifty in all. We were targeting mobiles at 720x1280 and desktops at 1080x1920. We moved elements around within all videos so they looked as good as we could make them on mobiles and desktops. Not perfect, but the 720's looked good on all mobiles we tested and the 1080's looked good in full screen even on my 43" 4k monitor. 

As it stands, our mobile videos look terrible on desktops and desktop videos look terrible on mobiles. For now, we have used the `<amp-story standalone` attribute to render the mobile videos on desktops within the AMP story carousel. Everyone I have shown them too on desktop prefers the full screen 1080 desktop experience we originally had over the 720 mobile in carousel version. Unless I'm missing something I think this change has rendered our 1080 desktop videos useless.

Sorry for sounding fed up. I put a lot of work and late nights into our videos. I'm surprised a bit of wizzardry WASM or something can't solve this but if you guys can't I guess nobody can.
